contracts/router/SynapseRouterV2.sol

```diff
@@ -0,0 +1,361 @@
+    function _getConnectedBridgeTokens(address token, bool origin)
+        internal
+        view
+        returns (BridgeToken[] memory connected)
+    {
+        uint256 len = _bridgeModules.length();
+        BridgeToken[][] memory unflattened = new BridgeToken[][](len);
+
+        uint256 count;
+        for (uint256 i = 0; i < len; ++i) {
+            (, address bridgeModule) = _bridgeModules.at(i);
+            BridgeToken[] memory bridgeTokens = IBridgeModule(bridgeModule).getBridgeTokens();
+
+            // assemble limited token format for quoter call
+            uint256 amountFound;
+            bool[] memory isConnected = new bool[](bridgeTokens.length);
+            for (uint256 j = 0; j < bridgeTokens.length; ++j) {
+                LimitedToken memory _tokenIn = origin
+                    ? LimitedToken({actionMask: ActionLib.allActions(), token: token}) // origin bridge tokens
+                    : LimitedToken({
+                        actionMask: IBridgeModule(bridgeModule).tokenToActionMask(bridgeTokens[j].token),
+                        token: bridgeTokens[j].token
+                    }); // dest bridge tokens
+                address _tokenOut = origin ? bridgeTokens[j].token : token;
+                isConnected[j] = swapQuoter.areConnectedTokens(_tokenIn, _tokenOut);
+                if (isConnected[j]) amountFound++;
+            }
+
+            // push to unflattened tokens if bridge token connected to given token
+            unflattened[i] = new BridgeToken[](amountFound);
+            count += amountFound;
+
+            uint256 k;
+            for (uint256 j = 0; j < bridgeTokens.length; ++j) {
+                if (isConnected[j]) {
+                    unflattened[i][k] = bridgeTokens[j];
+                    k++;
+                }
+            }
+        }
+
+        // flatten into connected tokens
+        connected = unflattened.flatten(count);
+    }
+
+    /// @notice Gets all bridge tokens for supported bridge modules
+    function _getBridgeTokens() internal view returns (BridgeToken[] memory) {
+        uint256 len = _bridgeModules.length();
+        BridgeToken[][] memory unflattened = new BridgeToken[][](len);
+
+        uint256 count;
+        for (uint256 i = 0; i < len; ++i) {
+            (, address bridgeModule) = _bridgeModules.at(i);
+            BridgeToken[] memory tokens = IBridgeModule(bridgeModule).getBridgeTokens();
+
+            // push to unflattened
+            unflattened[i] = new BridgeToken[](tokens.length);
+            count += tokens.length;
+
+            for (uint256 j = 0; j < tokens.length; ++j) {
+                unflattened[i][j] = tokens[j];
+            }
+        }
+
+        // flatten into bridge tokens array
+        return unflattened.flatten(count);
+    }
+
+    /// @notice Calculates amount of bridge token in accounting for bridge fees
```

Comment:
Let's add a note that it returns zero amount if fee is too big (rather than reverts)